lint.go

```diff
@@ -1027,34 +1027,47 @@ func (f *file) lintElses() {
 		if !ok || ifStmt.Else == nil {
 			return true
 		}
-		if ignore[ifStmt] {
-			return true
-		}
-		if elseif, ok := ifStmt.Else.(*ast.IfStmt); ok {
-			ignore[elseif] = true
-			return true
-		}
-		if _, ok := ifStmt.Else.(*ast.BlockStmt); !ok {
-			// only care about elses without conditions
+
+		if _, ok := ignore[ifStmt]; ok {
 			return true
 		}
+
+		outdentElse := true
+
+		// The else-block should only be outdented if the if block ends with a return statement.
```

Comment:
Is there are reason to detect 'return' but not other control statements such as 'continue' and 'break'?

Reply:
The original code handles only "returns". I guess it would be better to add "continue" and "break" in another PR unless you really want me to?